lib/topological_inventory/orchestrator/metric_scaler/prometheus_watcher.rb

```diff
@@ -75,12 +75,31 @@ def download_metrics
         end
 
         def scale_to_desired_replicas
-          super if scaling_allowed?
+          return unless configured?
+
+          desired_count = desired_replicas
+
+          return if desired_count == deployment_config.spec.replicas # already at max or minimum
+
+          logger.info("Scaling #{deployment_config_name} to #{desired_count} replicas")
+          object_manager.scale(deployment_config_name, desired_count)
+
+          # Wait for scaling to complete in Openshift
+          sleep(1) until worker_pods.length == desired_count
+          logger.info("Scaling #{deployment_config_name} complete")
+
           @scaling_allowed.value = false
         end
 
         private
 
+        def worker_pods
+          object_manager
+            .get_pods
+            .select { |pod| pod.metadata.annotations["openshift.io/deployment-config.name"] == deployment_config_name }
```

Comment:
Would this have to change later when we start using deployments instead of deployment-configs?

Reply:
> I'm not seeing, where we remove the code that was counting ips instead of Running pods?

The change is in the superclass, we were just calling `super` before and that was failing. 


> Would this have to change later when we start using deployments instead of deployment-configs?

Yes - this will get changed when we move orchestrator to use Deployments